From 066d61eceddb42365f0e386dac20ced013b4f0b7 Mon Sep 17 00:00:00 2001
From: Jiebing Li <jiebing.li@intel.com>
Date: Thu, 19 Mar 2015 17:06:58 +0100
Subject: [PATCH] usb: gadget: ffs: avoid disable ep twice

When FFS gadget function is disabled, gadget driver tries to
disable the endpoints twice, which triggers warning message
in device driver.

This patch ensures that usb_ep_disable() is not called again
after the endpoint has been disabled.

Change-Id: Ie2ad3b69112791d3e4b8a55d9943a38eee090692
Signed-off-by: Jiebing Li <jiebing.li@intel.com>
Signed-off-by: David Cohen <david.a.cohen@intel.com>
(cherry picked from commit b237fdddbd2ab64d6e84bf7cea259175535ee0fb)

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>
---
 drivers/usb/gadget/f_fs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/f_fs.c b/drivers/usb/gadget/f_fs.c
index 44cf775..d1640c9 100644
--- a/drivers/usb/gadget/f_fs.c
+++ b/drivers/usb/gadget/f_fs.c
@@ -1539,7 +1539,7 @@ static void ffs_func_eps_disable(struct ffs_function *func)
 	spin_lock_irqsave(&func->ffs->eps_lock, flags);
 	do {
 		/* pending requests get nuked */
-		if (likely(ep->ep))
+		if (ep->ep && epfile->ep)
 			usb_ep_disable(ep->ep);
 		epfile->ep = NULL;
 
-- 
1.7.9.5

